pyapp/data_collection.py

```diff
@@ -0,0 +1,105 @@
+import os
+from .data_obj import Node
+from prometheus_client import  Gauge, start_http_server
+from kubernetes import client, config
+from typing import Dict
+
+# starting the server so that promethus can read the data
+def start_server(port=8000)->None:
+    '''
+    create prometheus server with port number from env
+    '''
+    if "SEVERPORT" in os.environ:
+        port  = int(os.getenv('SEVERPORT'))
+    start_http_server(port)
+
+def create_guage():
+    '''
+    create guage metric for gpu
+    '''
+    # creating the guage metric for promethus 
+    # guage metric can increase and decrease check more on prometheus metric type
+    gpu_metric = Gauge('dsmlp_gpu', 'Number of GPUS in use',['hostname','type','status','course'])
+    return gpu_metric
+
+def combine_node_pod_data(node_data,pod_data)->Dict:
```

Comment:
list types for `node_data` and `pod_data`

Reply:
Consider renamed `combine_node_pod_data` into something like `build_nodes` or `to_nodes` since you're producing it. Also, you may want to specify the return type further from `Dict` to `Dict[str, Node]` for further clarification